Screens/Gutka.tsx

```diff
@@ -1,76 +1,54 @@
 import {
-  View, FlatList, StyleSheet,
+  View, StyleSheet,
 } from 'react-native';
 import { useTheme } from 'react-native-paper';
-import { useMainStoreState } from '../app_config/app_state/easy-peasy/hooks';
 
-import { loadShabad, loadBani } from '../app_config/database/banidb_api';
-import LineBlock from '../Components/Main/LineBlock';
 import ShimmeringLine from '../Components/Main/ShimmeringBlock';
 import Toolbar from '../Components/Main/Toolbar';
 import HighlightSelector from '../Components/Main/HighlightSelector';
 
 
 import { EditCtx } from '../app_config/app_state/easy-peasy/models';
 import { useValues } from '../app_config/app_state/state_hooks';
-import { mapModsToArray } from '../app_config/functions';
+import Viewer from '../Components/Main/Viewer';
+import { parseLines } from '../app_config/database/banidb_api';
 
 const Gutka = () => {
   const theme = useTheme();
+
   const [ shabads, updateShabads ] = useState( [] );
-  const [ dataLoading, updateLoading ] = useState( true );
   const [ isHighlighterVis, toggleHighligher ] = useState( false );
+  const [ isLoadingData, updateLoading ] = useState( true );
+
   const { isEditMode, selectedInfo } = EditCtx.useStoreState( ( store ) => ( { ...store } ) );
-  const { currentName, currentItems } = useValues( 'currentModel' );
-  const { baniLength } = useValues( 'viewerModel' );
+  const {
+    currentName, currentItems,
+  } = useValues( 'currentModel' );
+
+  const [ gutkaName ] = currentName;
 
-  const isDataReady = useMainStoreState(
-    ( store ) => store.gutkaModel.isDataReady,
-  );
 
   const { updateEditMode } = EditCtx.useStoreActions( ( actions ) => ( { ...actions } ) );
   useEffect( () => {
     updateLoading( true );
-  }, [ currentName[0] ] );
-
+  }, [ gutkaName ] );
   useEffect( () => {
     const getLines = async () => {
-      const numBanis = currentItems.reduce( ( count, { type } ) => ( type === 'Bani' ? count + 1 : 0 ), 0 );
-      if ( numBanis !== 0 ) {
-        // since banis take time to load, show this while its loading
-        updateLoading( true );
-      }
-      // if currentItems has a length greater than 0, get all the lines, otherwise set the array to empty
-      const newItems = currentItems
-        ? await Promise.all(
-          currentItems.map( ( item ) => ( item.type === 'Bani' ? loadBani( item.shabadId, baniLength ) : loadShabad( item.shabadId ) ) ),
-        )
-        : [ ];
-      updateShabads( newItems );
+      const newLines = currentItems.length ? await Promise.all( currentItems.map( ( item ) => parseLines( item ) ) ) : [];
+      updateShabads( newLines );
       updateLoading( false );
     };
-    if ( isDataReady && currentItems.length > 0 ) {
-      getLines();
-    } else if ( isDataReady && currentItems.length === 0 ) {
-      updateShabads( [] );
-      updateLoading( false );
-    }
-  }, [ currentItems, isDataReady, currentName[0] ] );
+    setTimeout( () => getLines(), 0 );
```

Comment:
So the way I can see this working under the hood:

- queue call to the defined version of `getLines` to the next tick. This may contain stale values/functions if any values that `getLines` uses are updated in the current tick by something else. Might be worth considering.

- React runs renders probably, followed by `getLines`, or the other way around, not sure.

The combination of stale values and execution order may be what makes the `setTimeout` hack work. Maybe two wrongs do make a right 😂.

Reply:
lol it used to work when it was just async but it wasnt so i just had it timeout lmao